ts_src/lib/interfaces.ts

```diff
@@ -1,16 +1,23 @@
+export type TransactionFromBuffer = (buffer: Buffer) => Transaction;
+export interface Transaction {
+  getInputOutputCounts(): { inputCount: number; outputCount: number };
+  addInput(objectArg: any): void;
+  addOutput(objectArg: any): void;
+  toBuffer(): Buffer;
+}
+
 export interface KeyValue {
   key: Buffer;
   value: Buffer;
 }
 
 export interface PsbtGlobal {
```

Comment:
It would make better sense to let base interface have `unknownKeyVals`.

```typescript
export interface PsbtMapBase {
  unknownKeyVals: KeyValue[];
}

export interface PsbtGlobal extends PsbtMapBase {
  // ...
}

export interface PsbtInput extends PsbtMapBase {
  // ...
}

export interface PsbtOutput extends PsbtMapBase {
  // ...
}
```